findbugs/src/java/edu/umd/cs/findbugs/detect/FindRefComparison.java

```diff
@@ -967,16 +969,32 @@ private void checkRefComparison(Location location, JavaClass jclass, Method meth
             if (lhsType.equals(Type.OBJECT) && rhsType.equals(Type.OBJECT)) {
                 return;
             }
-            String lhs = SignatureConverter.convert(lhsType.getSignature());
-            String rhs = SignatureConverter.convert(rhsType.getSignature());
 
-            if ("java.lang.String".equals(lhs) || "java.lang.String".equals(rhs)) {
+            String lhsSignature = lhsType.getSignature();
+            @SlashedClassName
+            String lhsSlashed = ClassName.extractClassName(lhsSignature);
+            @DottedClassName
+            String lhsDotted = SignatureConverter.convert(lhsSignature);
+
+            String rhsSignature = rhsType.getSignature();
+            @SlashedClassName
+            String rhsSlashed = ClassName.extractClassName(rhsSignature);
```

Comment:
I hope the errors you report are gone now.

> Running ant in findbugsTestCases should not report any errors, nor any missing classes.

Running `ant` causes all kinds of errors and I must admit that I am hesitant to invest time into solving them - getting to a state where I can work with FindBugs was a little tedious and I am happy that it works at all.